kornia/feature/disk/disk.py

```diff
@@ -0,0 +1,121 @@
+from typing import List, Optional, Tuple
+
+import torch
+
+from kornia.core import Tensor
+from kornia.utils.helpers import map_location_to_cpu
+
+from .detector import heatmap_to_keypoints
+from .structs import DISKFeatures
+from .unets import Unet
+
+
+class DISK(torch.nn.Module):
+    r"""Module which detects and described local features in an image using the DISK method. See
+    :cite:`tyszkiewicz2020disk` for details.
+
+    Args:
+        desc_dim: The dimension of the descriptor.
+        unet: The U-Net to use. If None, a default U-Net is used. Kornia doesn't provide the training code for DISK
+              so this is only useful when using a custom checkpoint trained using the code released with the paper.
+              The unet should take as input a tensor of shape :math:`(B, C, H, W)` and output a tensor of shape
+              :math:`(B, \mathrm{desc\_dim} + 1, H, W)`.
+
+    Example:
+        >>> disk = DISK.from_pretrained('depth')
+        >>> images = torch.rand(1, 3, 256, 256)
+        >>> features = disk(images)
+    """
+
+    def __init__(self, desc_dim=128, unet=None):
+        super().__init__()
+
+        self.desc_dim = desc_dim
+
+        if unet is None:
+            unet = Unet(in_features=3, size=5, down=[16, 32, 64, 64, 64], up=[64, 64, 64, desc_dim + 1])
+        self.unet = unet
+
+    def heatmap_and_dense_descriptors(self, images: Tensor) -> Tuple[Tensor, Tensor]:
+        """Returns the heatmap and the dense descriptors."""
+
+        try:
+            unet_output = self.unet(images)
+        except RuntimeError as e:
+            if 'Trying to downsample' in str(e):
+                raise ValueError(
+                    f'The current implementation of U-Net requires image size to '
+                    f'be divisible by 16 (got {images.shape[2:]}).'
+                )
+            else:
+                raise
```

Comment:
we can't test it before forward unet? to avoid a unecessary try-except here

Reply:
Good point.